hermes/hermes.go

```diff
@@ -0,0 +1,103 @@
+}
+
+// StateJournal stores the state of Hermes in two parts: the next operation intent and a map of filenames.
+// The intent stores the operation to be performed and the name of the file that the operation is being performed on.
+// The filenames map is a map of file IDs to filenames.
+// If an entry does not exist for a given ID, then the file does not exist (i.e. has been deleted)
+type StateJournal struct {
+	intent    Intent         // intent stores the next intended file operation and the name of the file that the operation is being performed on.
+	filenames map[int]string // filenames is a map of file IDs to filenames.
+}
+
+// Init initialises the map in the StateJournal so that entries can be added to it.
+func (sj *StateJournal) Init() {
+	sj.filenames = make(map[int]string) // initialise filenames to a map
+}
+
+// Hermes is the main Hermes prober that will startup Hermes and initiate monitoring targets.
+type Hermes struct {
+	Journal           StateJournal    // stateJournal stores the state of Hermes as a combination of next operation intent and a filenames map
+	Ctx               context.Context // Context for starting Cloudprober
+	CancelCloudprober func()          // cancelCloudprober is a cancel() function associated with the context passed to Cloudprober when initialised.
+}
+
+// InitialiseCloudproberFromConfig initialises Cloudprober from the config passed as an argument.
+// Parameters:
+// - config: config should be the contents of a Cloudprober config file. This is most likely: "grpc_port=9314"
+//           -> the "grpc_port:" field is the only required field for the config.
+// Returns:
+// - error:
+//	- logger.NewCloudproberLog() error: error initialising logging on GCE (Stackdriver)
+//	- sysvars.Init():
+//		- error getting local hostname: [error]:
+//			-> error getting hostname from os.Hostname()
+//		- other error
+//			-> error initialising Cloud metadata
+//	- config.ParseTemplate() error:
+//		-> regex compilation issue of config or config could not be processed as a Go text template
+//	- proto.UnmarshalText() error:
+//		-> The config does not match the proto that it is being unmarshalled with.
+//	- initDefaultServer() error:
+//		- failed to parse default port from the env var: [serverEnvVar]=[parsedPort]
+//		- error while creating listener for default HTTP server: [error]
+//	- error while creating listener for default gRPC server: [error]
+//	- tlsconfig.UpdateTLSConfig() error: an error occurred when updating the TLS config from the config passed.
+func (hermes *Hermes) InitialiseCloudproberFromConfig(config string) error {
+	err := cloudprober.InitFromConfig(config)
+	if err != nil {
+		glog.Errorf("failed to initialise cloudprober, err: %v", err)
+		return err
+	}
+
+	hermes.Ctx, hermes.CancelCloudprober = context.WithCancel(context.Background()) // Create new context with a cancel() function stored in the Hermes struct
```

Comment:
Move the inline comment to the line above.
```suggestion
        // Create new context with a cancel() function stored in the Hermes struct
	hermes.Ctx, hermes.CancelCloudprober = context.WithCancel(context.Background())
```

Reply:
I've made this change.